sky/resources.py

```diff
@@ -920,12 +920,20 @@ def _try_validate_ports(self) -> None:
         """
         if self.ports is None:
             return
-        if skypilot_config.get_nested(('aws', 'security_group_name'),
-                                      None) is not None:
-            with ux_utils.print_exception_no_traceback():
-                raise ValueError(
-                    'Cannot specify ports when AWS security group name is '
-                    'specified.')
+        if self.cloud is None or isinstance(self.cloud, clouds.AWS):
+            security_group_name = skypilot_config.get_nested(
+                ('aws', 'security_group_name'), None)
+            if security_group_name is not None:
+                with ux_utils.print_exception_no_traceback():
+                    logger.warning(
+                        f'Ports {self.ports} and security group name are '
+                        f'specified: {security_group_name}. It is not '
+                        'guaranteed that the ports will be opened if the '
+                        'specified security group is not correctly set up. '
+                        'Please try to specify `ports` only and leave out '
+                        '`aws.security_group_name` in `~/.sky/config.yaml` to '
+                        'allow SkyPilot to automatically create and configure '
+                        'the security group.')
```

Comment:
I'm not sure this is the ideal place as it seems to print the warning `10+` times per launch.

Reply:
This is a good catch! Let's remove the warning here and move it to the `aws.make_deploy_resources_variables` so that the warning will only shown when launching an exact launchable resource. (check out the comment in aws.py above)